### ISSUE TITLE
Angular Universal with PWA is not google search crawl friendly

### ISSUE THREAD

**Issue by muthufmass** [angular/universal#1113]
# Bug Report

#### What is the expected behavior?
Meta data to be crawled and page to be SEO enabled.

#### What is the current behavior?
Site is server side rendered now using angular universal, able to see all metadata on the head and site rendered smooth on the chrome browser. But the same when we try from 'Google Search Console' with Fetch as Google , site is not loading and no meta data is coming up. Hence even after moving to seo friendly angular universal, still struck on meta crawling issues and seo friendly page.

#### What modules are related to this issue?
```
- [ ] common
- [ ] express-engine
- [ ] module-map-ngfactory-loader
```

#### Minimal reproduction with instructions:
https://sporfy.com/home

Fetch as google in search console & render as a webpage in browser.

#### What is the use-case or motivation for changing an existing behavior?
Support search crawling

#### Environment:
**@nguniversal versions**
  - common: "7.1.3"
  - @nguniversal/express-engine: "^7.0.2"
  - @nguniversal/module-map-ngfactory-loader: "^7.0.2"

```
Output from: `ng --version`.
Angular CLI: 7.1.2
Node: 11.2.0
OS: darwin x64
Angular: 7.1.3
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.8.8
@angular-devkit/build-angular     0.11.2
@angular-devkit/build-optimizer   0.11.2
@angular-devkit/build-webpack     0.11.2
@angular-devkit/core              7.1.2
@angular-devkit/schematics        7.1.2
@angular/cdk                      7.1.1
@angular/cli                      7.1.2
@angular/fire                     5.1.1
@angular/material                 7.1.1
@angular/pwa                      0.11.2
@ngtools/webpack                  7.1.2
@schematics/angular               7.1.2
@schematics/update                0.8.8
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1
```

#### Is there anything else we should know?
I could be wrong on implementation, then need guidance in fixing the same. before universal movement, and just as SPA app, it was not even loading all the metadata, now on inspect, we find the page is rendering properly in inspect element view with all metas. But when I go to view page source, still finding issues on page rendered content.

**Comment by D2KX** [comment]
![sporfy](https://user-images.githubusercontent.com/4834966/51076622-0324dc00-169b-11e9-8060-e2f7223149cb.png)

Yeah, there is no metatags coming from the server side rendered document (however, basic server side rendering seems to be working, as the document is quite large and looks a lot like what you would expect from Angular Universal). This is likely a minor implementation issue and we would have to know the code/at what point/how the metatags are set up in the application's code.

However I also noticed that the bundle size and overall size of the site is extremely large, and there were multiple errors thrown in the browser's console, so it could also very well be that there are some other issues that maybe prevent your meta-tag logic from properly running. I would not be too surprised if your Node.js server running Angular Universal would throw a lot of (useful) error logs upon each visit right now, and I would look into lazy-loading/trying to get the bundle size of your JavaScript and the asset size of your images down on first visit.

**Comment by muthufmass** [comment]
We have managed to get the meta tags now. We switched off the service worker, as we were trying previously with PWA + angular universal. We are improved in reducing our bundle size but still not great.

we need certain npm plugins used like bootstrap, jquery, chart components etc, which we cannot remove, but bundle sizes are some thing of a concern being very large. Any recommendations with universal ? In other webpack projects, we have capacity to split the bundles, here with angular.json its restrictive on bundle splits.
